test/integration/app-types/src/app/type-checks/link/page.tsx

```diff
@@ -62,7 +62,7 @@ export default function page() {
       <Link href="/redirect">test</Link>
       <Link href="/redirect/v1/guides/x/page">test</Link>
       <Link href="/redirect/guides/x/page">test</Link>
-      <Link href={new URL('/about')}>test</Link>
+      <Link href={{ ...new URL('https://nextjs.org') }}>test</Link>
```

Comment:
Why did this PR change what's supported in `Link`?

Reply:
ah, this PR didn't change it, i changed it because if you tried to run this page, it was erroring with that "cannot pass a URL to a client component, RSC only allows plain objects" message. i guess we could also add a `'use client'` at the top instead, do you think that'd be better?

Reply:
I'd say so since this file is about types primarily. Right now it looked like we'd no longer allow URL in `Link`

Reply:
alright, i split this change into a separate PR just to not muddy the waters here
https://github.com/vercel/next.js/pull/68360